system-int-test/test/cmr/system_int_test/bootstrap/bulk_index_test.clj

```diff
@@ -690,3 +694,85 @@
       (testing "All tag parameters with XML references"
         (is (d/refs-match? [coll1]
                            (search/find-refs :collection {:tag-key "tag1"})))))))
+
+(deftest bulk-index-variables-for-provider
+  (testing "Bulk index variables for a single provider"
+    (s/only-with-real-database
+      ;; Disable message publishing so items are not indexed.
+      (dev-sys-util/eval-in-dev-sys
+       `(cmr.metadata-db.config/set-publish-messages! false))
+
+      ;; The following is saved, but not indexed due to the above call
+      (let [var1 (variable/ingest-variable-with-attrs {:provider-id "PROV1"} {} 1)]
+        (is (= 0 (:hits (variable/search {}))))
+        (bootstrap/bulk-index-variables "PROV1")
+        (index/wait-until-indexed)
```

Comment:
Oh right - because we are using the message queue we actually can test this without forcing to be synchronous. Another advantage of moving away from core.async.